static/js/components/google_map.tsx

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for embedding a map from Google Maps
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React from "react";
+
+const DEFAULT_MAP_ZOOM = 4;
+
+// number of digits after decimal point in LatLong/<lat>_<long> dcids
+const LAT_LONG_PRECISION = 5;
+
+interface GoogleMapPropType {
+  /**
+   * DCID of the place/event to show a map for.
+   */
+  dcid: string;
+}
+
+interface GoogleMapStateType {
+  /**
+   * Stores responses from api calls.
+   */
+  markerLocation: GoogleMapCoordinates;
+  mapInfo: MapInfoResponse;
+}
+
+interface GoogleMapCoordinates {
+  /**
+   * Lat/Long coordinates, stored in a form the Google maps API expects.
+   */
+  lat: number;
+  lng: number;
+}
+
+interface MapInfoResponse {
+  /**
+   * Response format of /api/place/mapinfo/.
+   */
+  up: number;
+  down: number;
+  left: number;
+  right: number;
+  coordinateSequenceSet: google.maps.LatLng[][];
+}
+
+/**
+ * Converts a DCID of the form latLong/<latitude>_<longitude> to
+ * lat/long coordinates used by the google maps API.
+ * @param dcid latLong DCID to convert
+ * @returns a {lat: latitude, lng: longitude} object for the google maps API
+ */
+function convertLatLongDcid(dcid: string): GoogleMapCoordinates {
+  if (dcid.startsWith("latLong/") && dcid.includes("_")) {
+    const coordinateStrings = dcid.slice("latLong/".length).split("_");
+    const coords = coordinateStrings.map((coordString) =>
+      // Add decimal point to parsed latitude and longitude numbers
+      parseFloat(
+        coordString.slice(0, -LAT_LONG_PRECISION) +
```

Comment:
please add lots of comments for these (and best if we can reference the backend equivalent which needs to stay in sync). or vice versa, depending on where the code lives.

Reply:
Instead of having to make sure the client code stays in sync with the backend code, should we just always read the property values of "latitude" and "longitude" .. I don't think fetching that data would significantly impact speed or memory usage?

Reply:
Went with chejennifer's suggestion of always reading lat/long for property values. It feels cleaner, and avoids potential mismatch with backend code. This section was removed.